Packs/VMwareESXi/ModelingRules/VMwareESXiModelingRules.xif

```diff
@@ -0,0 +1,87 @@
+[MODEL: dataset="VMware_ESXi_raw", model=Endpoint]
+ filter _raw_log contains "hostd" or _raw_log contains "vcpu" or _raw_log contains "svga"  or _raw_log contains "mks" or _raw_log contains "vmx" 
+| filter  _raw_log not contains "verbose" 
+|alter request_time = arrayindex(regextract(_raw_log ,"(\d+\-\d+\-\d+T\d+\:\d+\:\d+Z?\.?\d+Z)"),0)
+|alter service_name= arrayindex(regextract(_raw_log ,"\d+\-\d+\-\d+T\d+\:\d+\:\d+\S\d+\w+\s?\w+\|?\s([^\[?\|?\]]+)") ,0)
+// extract message
+| alter message1= arrayindex(regextract(_raw_log , "\|\s\w+\:\s([^\"]+)") ,0)
+| alter message2 = arrayindex(regextract(_raw_log , "opID\=[^\s]+\s([^\"]+)"),0)
+| alter message = coalesce(message2 ,message1)
+// end extract message
+| alter opID= arrayindex(regextract(_raw_log ,"opID\=([^\]]+)") ,0)
+| alter originator=arrayindex(regextract(_raw_log ,"Originator\@(\d+)") ,0)
+// extract username
+| alter user= arrayindex(regextract(_raw_log ,"user\:\s(\w+)") ,0)
+| alter vpxuser= arrayindex(regextract(_raw_log ,"vpxuser\@(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"),0)
+| alter username = coalesce(user ,vpxuser)
+// end extract user
+| alter sub = arrayindex(regextract(_raw_log ,"sub\=([^\s]+)") ,0)
+| alter XDM.Endpoint.operation =service_name 
+| alter XDM.Endpoint.operation =opID
+| alter XDM.Endpoint.Observer.unique_identifier =originator
+| alter XDM.Endpoint.original_event_sub_type= sub
+| alter XDM.Endpoint.Actor.user.identifier  =username
+| alter XDM.Endpoint.event_timestamp= parse_timestamp("%b %d %H:%M:%S",request_time)
+| alter XDM.Endpoint.threat.description=message;
+
+
+[MODEL: dataset="VMware_ESXi_raw", model=Audit]
+filter _raw_log contains "verbose" or _raw_log contains "vpax" or _raw_log contains "scsiCorrelator" or _raw_log contains "cpu" or _raw_log contains "mark" or _raw_log contains "vmkeventd" or _raw_log contains "vmkdevmgr" or _raw_log contains "lwsmd" or _raw_log contains "smrtd" or _raw_log contains "sysboot" or _raw_log contains "ESXShell" or _raw_log contains "SSH" or _raw_log contains "mark" or _raw_log contains "esxupdate" or _raw_log contains "dhclient-um" 
+// extract message
+| alter message1= arrayindex(regextract(_raw_log ,"opID\=[^\s]+\s([^\"]+)") ,0)
+| alter message2 = arrayindex(regextract(_raw_log ,"\:\s([^\&]+)"),0)
+| alter message = coalesce(message2 ,message1)
+// end extract message
+| alter boot_timestamp= arrayindex(regextract(_raw_log ,"\[(\d+\-\d+\-\d+\s\d+\:\d+\:\d+\.\d+\s\-\s\d+\-\d+\-\d+\s\d+\:\d+\:\d+\.\d+)") ,0)
```

Comment:
need to change the rows 35-36 to:

`// extract timestamp part
| alter boot_timestamp= arrayindex(regextract(_raw_log ,"\[(\d+\-\d+\-\d+\s+\d+\:\d+\:\d+\.\d+)") ,0)
    ,extract_timestamp2= arrayindex(regextract(_raw_log ,"(\d+\-\d+\-\d+T\d+\:\d+\:\d+)") ,0)
| alter timestamp2 = concat(extract_timestamp2 ,".000000")
    ,boot_timestamp =replex(boot_timestamp , " ","T")
| alter Timestamp = coalesce(boot_timestamp , Timestamp2)
| alter the_timestamp = parse_timestamp("%Y-%m-%dT%H:%M:%E6S",Timestamp )
// end extarct timestamp`